syntax = "proto3";
option java_package = "godot.wire";

message Vector2 {
  float x = 1;
  float y = 2;
}

message Rec2 {
  Vector2 position = 1;
  Vector2 size = 2;
}

message Vector3 {
  float x = 1;
  float y = 2;
  float z = 3;
}

message KVariant {
  oneof type {
    int32 nil_value = 1;
    int64 long_value = 2;
    double real_value = 3;
    string string_value = 4;
    bool bool_value = 5;
    Vector2 vector2_value = 6;
    Rec2 rec2_value = 7;
    Vector3 vector3_value = 8;
  }
}

message KReturnValue {
  KVariant data = 1;
}

message KFuncArgs {
  repeated KVariant args = 1;
}